src/platform_impl/stdweb/event_loop.rs

```diff
@@ -0,0 +1,468 @@
+        match &*elrs.0.runner.borrow() {
+            Some(ref runner) if runner.control.is_exit() => return,
+            _ => ()
+        }
+
+        event.prevent_default();
+        event.stop_propagation();
+        event.cancel_bubble();
+
+        handler(&elrs, event);
+    });
+}
+
+impl<T: 'static> EventLoopRunnerShared<T> {
+    // Set the event callback to use for the event loop runner
+    // This the event callback is a fairly thin layer over the user-provided callback that closes
+    // over a RootEventLoopWindowTarget reference
+    fn set_listener(&self, event_handler: Box<dyn FnMut(Event<T>, &mut ControlFlow)>) {
+        *self.0.runner.borrow_mut() = Some(EventLoopRunner {
+            control: ControlFlowStatus::Init,
+            is_busy: false,
+            event_handler,
+        });
+        self.send_event(Event::NewEvents(StartCause::Init));
+    }
+
+    // Add an event to the event loop runner
+    //
+    // It will determine if the event should be immediately sent to the user or buffered for later
+    pub fn send_event(&self, event: Event<T>) {
+        // If the event loop is closed, it should discard any new events
+        if self.closed() {
+            return;
+        }
+
+        // Determine if event handling is in process, and then release the borrow on the runner
+        match *self.0.runner.borrow() {
+            Some(ref runner) if !runner.is_busy => {
+                let (start_cause, event_is_start) = if let Event::NewEvents(cause) = event {
+                    (cause, true)
+                } else {
+                    (match runner.control {
+                        ControlFlowStatus::Init => StartCause::Init,
+                        ControlFlowStatus::Poll { ref timeout } =>  {
+                            timeout.clear();
+
+                            StartCause::Poll
+                        }
+                        ControlFlowStatus::Wait { start } => StartCause::WaitCancelled {
+                            start,
+                            requested_resume: None,
+                        },
+                        ControlFlowStatus::WaitUntil { start, end, ref timeout } => {
+                            timeout.clear();
+
+                            StartCause::WaitCancelled {
+                                start,
+                                requested_resume: Some(end)
+                            }
+                        },
+                        ControlFlowStatus::Exit => { return; }
+                    }, false)
+                };
+                let mut control = runner.control.to_control_flow();
+                // Handle starting a new batch of events
+                //
+                // The user is informed via Event::NewEvents that there is a batch of events to process
+                // However, there is only one of these per batch of events
+                self.handle_event(Event::NewEvents(start_cause), &mut control);
```

Comment:
Calling this here after borrowing the `runner` causes a panic on my end. I think it's because `handle_event` wants to borrow it again mutably.

Reply:
Thanks for the heads up, should be fixed now.

The work that's left is logging, adding the onclose event once it's merged into stdweb, adding fullscreen mode and pointer capture support to stdweb, and adding support for those to this backend once they're merged into stdweb. If you want to tackle any of those, feel free!